Libplanet/Blockchain/BlockChain.cs

```diff
@@ -83,8 +83,15 @@ public Block<T> Tip
             get; private set;
         }
 
+        /// <inheritdoc/>
+        int IReadOnlyCollection<Block<T>>.Count =>
+            checked((int)Store.CountIndex(Id.ToString()));
+
         internal IStore Store { get; }
 
+        /// <inheritdoc/>
+        public Block<T> this[int index] => this[(long)index];
```

Comment:
@earlbread Although it's late, we should make it throw [`ArgumentOutOfRangeException`][1] instead of [`IndexOutOfRangeException`][2], because collection types in [`System.Collections.Generic`][3] throw `ArgumentOutOfRangeException` and only arrays throw `IndexOutOfRangeException`.  See also [this SO answer][4]:

> Note that `List<T>` throws `ArgumentOutOfRangeException` for the same cases where arrays use `IndexOutOfRangeException`.

Could you change this in the next patch?

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception
[2]: https://docs.microsoft.com/en-us/dotnet/api/system.indexoutofrangeexception
[3]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic
[4]: https://stackoverflow.com/a/20940980/383405

Reply:
Sure, I'll change it in the next patch!